src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -363,13 +363,13 @@ private void Initialize()
                 }
 
                 int fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
-                if (fileDescriptor < 0)
+                if (fileDescriptor != -1)
```

Comment:
are you saying that the result you are getting back from open and mmap is negative but that is still a success case? After making this change, where you able to use this driver to control pins in a raspberry pi 4?

Reply:
I am saying that the value I am getting back for a file descriptor is better compared to -1 as that is what the documentation says.

 This wasn't causing the error though … The subsequent mmap  was returning a value of 0xB5302000 when I tested it and it was then pushed through `mapPointer.ToInt32()` thus giving a negative number and failing the test. (hence the change to now test against -1)

I have not yet tested this from the point of view of driving pins I have just tested that the driver gets initialized without an IO error.
…

Reply:
**Oops. I see why the tests are failing …. Will send an amended PR**

Reply:
Yes, Per http://man7.org/linux/man-pages/man2/open.2.html

> open(), openat(), and creat() return the new file descriptor, or -1
> if an error occurred (in which case, errno is set appropriately).

this seems correct